packages/components/src/components/hds/tabs/index.ts

```diff
@@ -8,18 +8,40 @@ import { tracked } from '@glimmer/tracking';
 import { action } from '@ember/object';
 import { assert, warn } from '@ember/debug';
 import { next, schedule } from '@ember/runloop';
+import type { ComponentLike } from '@glint/template';
+import type { HdsTabsPanelSignature } from './panel';
+import type { HdsTabsTabSignature } from './tab';
+import type { HdsTabsSizeValues } from './types';
 
 export const DEFAULT_SIZE = 'medium';
 export const SIZES = ['medium', 'large'];
 
-export default class HdsTabsIndexComponent extends Component {
-  @tracked tabNodes = [];
-  @tracked tabIds = [];
-  @tracked panelNodes = [];
-  @tracked panelIds = [];
+interface HdsTabsIndexSignature {
+  Args: {
+    isParentVisible?: boolean;
+    selectedTabIndex?: number;
+    size?: HdsTabsSizeValues;
+    onClickTab?: (event: MouseEvent, tabIndex: number) => void;
+  };
+  Blocks: {
+    default: [
+      {
+        Panel?: ComponentLike<HdsTabsPanelSignature>;
+        Tab?: ComponentLike<HdsTabsTabSignature>;
+      }
+    ];
+  };
+  Element: HTMLDivElement;
+}
+
+export default class HdsTabsIndexComponent extends Component<HdsTabsIndexSignature> {
+  @tracked tabNodes: HTMLButtonElement[] = [];
+  @tracked tabIds: string[] = [];
+  @tracked panelNodes: HTMLElement[] = [];
+  @tracked panelIds: string[] = [];
```

Comment:
maybe?
```suggestion
  @tracked panelIds: HdsTabsPanelSignature['Args']['panelId'] = [];
```

Reply:
What do you think instead about defining a type like

```
type PanelIdList = string[]

...

@tracked panelIds: PanelIdList = [];
```

And then importing that type in panels and using it?

It feels weird to me to have a parent property type defined by a child component attribute.



````